gulpfile.js

```diff
@@ -246,21 +246,21 @@ gulp.task('ext:test', (done) => {
 	var args = ['--verbose', '--disable-gpu', '--disable-telemetry', '--disable-updates', '-n'];
 	let vscodeVersion = packageJson.engines.vscode.slice(1);
 	let extensionTestsPath = `${workspace}/out/test`;
-	vscodeTest.downloadAndUnzipVSCode(vscodeVersion).then((vscodePath) => {
-		if (vscodePath) {
-			vscodeTest.runTests({
-				vscodeExecutablePath: vscodePath,
-				extensionDevelopmentPath: workspace,
-				extensionTestsPath: extensionTestsPath,
-				launchArgs: args
-			}).then(() => done()).catch((error) => {
-				console.log(`stdout: ${process.stdout}`);
-				console.log(`stderr: ${process.stderr}`);
-				console.error(`exec error: ${error}`);
-				done(error);
-			});
-		}
-	})
+	let vscodePath = await vscodeTest.downloadAndUnzipVSCode(vscodeVersion);
+	try {
+		await vscodeTest.runTests({
+			vscodeExecutablePath: vscodePath,
+			extensionDevelopmentPath: workspace,
+			extensionTestsPath: extensionTestsPath,
+			launchArgs: args
+		});
+	} catch (error) {
+		console.log(`stdout: ${process.stdout}`);
+		console.log(`stderr: ${process.stderr}`);
+		console.error(`exec error: ${error}`);
```

Comment:
Shouldn't we fail this task if the tests fail?

Reply:
The task does fail because of inner assertion errors etc

Reply:
What do you mean "inner assertion errors"? How is the gulp task being notified of these errors if we're catching everything here?

Reply:
It's handled by the vscode-test module. The original idea was to capture all the errors and not just fail after one error, I just refactored it.

Reply:
I assume the "not fail after one error" was you removing the original `done(error)` call? But right now I'm not seeing you ever call done with an error which seems to me to imply that this task will never fail. Did you test this with purposely making a test fail to ensure that it fails the task if a test fails?

Reply:
yes, I tested it and it ends with an error and prints out the test with the error (exit code is 1 in that case)